### PR TITLE
Add charts for climate visualisation

### LINKED ISSUE #27: Charts here instead
Try adding the stuff for https://github.com/BHoM/Python_Toolkit/issues/21 to this Toolkit instead.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #27 

### Test files
[Charts.zip](https://github.com/BHoM/MachineLearning_Toolkit/files/4971721/Charts.zip)

You'll want to change the file locations obviously.

### Additional comments
**Make sure you run this against `master` branch of Python_Toolkit!**

**Run the `InstallPythonToolkit` component, then close Rhino, then run the `InstallMachineLearningToolkit`, then 
close Rhino, then try the test script**

### CODE REVIEW

File: MachineLearning_Engine/Compute/ClimateCharts/UTCI.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    # Reshape data into time/day matrix for heatmap plotting
+    annual_matrix = df.pivot_table(columns=df.index.date, index=df.index.time).values[::-1]
+
+    # Create UTCI colormap
+    utci_cmap = ListedColormap(['#0D104B', '#262972', '#3452A4', '#3C65AF', '#37BCED', '#2EB349', '#F38322', '#C31F25', '#7F1416', '#580002'])
+    utci_cmap_bounds = [-100, -40, -27, -13, 0, 9, 26, 32, 38, 46, 100]
+    utci_cmap_norm = BoundaryNorm(utci_cmap_bounds, utci_cmap.N)
+
+    bounds = np.arange(-41, 48, 1)
+    norm = BoundaryNorm(bounds, utci_cmap.N)
```
Comment (epignatelli) [inline MachineLearning_Engine/Compute/ClimateCharts/UTCI.py]: This breaks - the number of colors should be at least the same number as `len(bounds)`
```suggestion
    norm = BoundaryNorm(bounds, max(len(bounds), utci_cmap.N))
```

**Comment by BingWangUS** [comment]
Tested with today's installer, installed toolkits as description, still having the issue "1. AttributeError : type object 'pandas._libs.tslibs.conversion._TSObject' has no attribute '__reduce_cython__'" when import ML_Engine.Compute

**Comment by michaelhoehn** [comment]
> Tested with today's installer, installed toolkits as description, still having the issue "1. AttributeError : type object 'pandas._libs.tslibs.conversion._TSObject' has no attribute '**reduce_cython**'" when import ML_Engine.Compute

+1 on this process and error. Same Invoke error on plot methods as before on my machine.

**Comment by vbeardow** [comment]
@tg359 and I had a play around this morning and managed to get the charts working in Rhino 5. Same errors as above were found when trying to use charts/import pandas with Rhino 6!

**Review by michaelhoehn — DISMISSED** [review]
Approved on the understanding that this solution works only in Rhino 5 and with some occasional, undesirable behavior. I could only produce the Windrose and UTCI charts successfully (I also tried Diurnal but with no luck). Components error out somewhat unpredictably and the canvas has to be recomputed in order to generate additional plots. When the error occurs, it is still on Invoke method.

I think this is a great setting out point and really looking forward to a few functionality fixes.

**Comment by epignatelli** [comment]
Thanks very much guys for the reviews.
 
I would say, let's get to a point where the solution is stable enough - it seems like there are still some issues.

I'll try investigate the pandas and the Rhino 6 issue this week. 
@tg359 can you make sure the python code is bug-free with the versions of the packages we have in the BHoM?
I know `windrose` will fail with the current version of `matplotlib`.
It needs manual installation `pip install git+...`, but not everyone has git installed on his/her machine, this will fail.
Is there another solution there?

**Review by epignatelli — APPROVED** [review]
Alright guys, thanks very much for gathering around this.
I think this is ready to be merged with some assumptions to be kept in mind.

- We removed the windrose chart because it relies on the `windrose` package, which requires git to be installed. That's not safe enough, we can't rely on that. Better find another way to deal with it.
STATUS: Issue not raised yet.

There are some individial issues with the python code. STATUS: Issues raised:
- #34 
- #35
- #36

Plotting works only on Rhino 5. STATUS: Issue raised:
- #37

I'll make sure with @FraserGreenroyd that the toolkit is included in the installer after this pr is merged.

Thanks guys - great one in! <3

**Pull request merged**